pylearn2/datasets/dataset.py

```diff
@@ -48,6 +48,28 @@ def iterator(self, mode=None, batch_size=None, num_batches=None,
             through the dataset and may potentially be shared by
             multiple iterator objects simultaneously (see "Notes"
             below).
+        data_specs : (space, source) pair, optional
+            `space` is an instance of `Space` (possibly a
+            `CompositeSpace`), and `source` is a string (or tuple of
+            strings if `space` is a `CompositeSpace`). `source` defines
+            which data should be returned by the iterator (for instance
+            'features', 'targets'), and `space` specifies in which
+            format it will be returned. These specifications should be
+            flat: if `space` is a `CompositeSpace`, the sub-spaces should
+            not be `CompositeSpace`s, and `source` should not be a nested
+            tuple, and the same sub-source should not be mentioned more
+            than once with the same sub-space. If `data_specs` is not
```

Comment:
"the same sub-source should not be mentioned more than once with the same sub-space"
By "with" do you mean "within"?
Isn't the global Space technically also a sub-space, and doesn't that imply that no sub-source  can be repeated?
Since the specification must be flat, doesn't that mean that `source` should be a list of unique strings?
It seems much simpler to say that source should be a list of unique strings than to have all this stuff about sub-spaces and sub-sources.